nvflare/private/fed/server/server_json_config.py

```diff
@@ -143,10 +171,49 @@ def process_config_element(self, config_ctx: ConfigContext, node: Node):
             if cid in self.components:
                 raise ConfigError('duplicate component id "{}"'.format(cid))
 
-            self.workflows.append(WorkFlow(cid, workflow))
-            self.components[cid] = workflow
+            workflow = WorkFlow(cid, responder)
+            self.workflows.append(workflow)
+            self.components[cid] = responder
             return
 
+    def get_responder(self, component):
+        if isinstance(component, dict):
+            wf_config = component
+            communicator = wf_config.get(CommConstants.COMMUNICATOR)
+            if communicator is None:
+                communicator = WFCommunicator()
+
+            if isinstance(communicator, WFCommunicatorSpec):
+                controller_config = wf_config.get(CommConstants.CONTROLLER)
+                controller_config["lazy_instantiate"] = False
+                communicator.set_controller_config(controller_config)
+            data_bus = DataBus()
+            data_bus.put_data(CommConstants.COMMUNICATOR, communicator)
```

Comment:
what if there are multiple workflows with communicator? what will happen?

Reply:
the later communicator will just overwrite the former communicator which seems incorrect? maybe we need to use a different key here?

Reply:
Good catch, appended workflow communicator "id" key as different workflows can use different communicators.